client/src/app/components/deadline-counter/deadline-counter.component.ts

```diff
@@ -33,17 +34,20 @@ export class DeadlineCounterComponent implements OnInit {
     }
     private startCounter(){
         this.interval = setInterval(()=>{
-            const now = new Date().getTime() - 100000;
-            const countDownDate = this.date.getTime();
-            var distance = countDownDate - now;
-            this.daysLeft = Math.floor(distance / (1000 * 60 * 60 * 24));
-            this.hoursLeft= Math.floor((distance % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60));
-            this.minutesLeft = Math.floor((distance % (1000 * 60 * 60)) / (1000 * 60));
-            this.secondsLeft = Math.floor((distance % (1000 * 60)) / 1000);
-            if(distance < 0){
+            const now = moment();
+            const date = moment(this.date);
+            this.daysLeft = date.diff(now, 'days');
+            this.hoursLeft = date.diff(now.add(Number(this.daysLeft), 'days'), 'hours');
+            this.minutesLeft = date.diff(now.add(Number(this.hoursLeft), 'hours'), 'minutes');
+            this.secondsLeft = date.diff(now.add(Number(this.minutesLeft), 'minutes'), 'seconds');
+            if(date.diff(now) < 0){
```

Comment:
You can simplify it even further, by combining duration and diff, like:
```
const duration = moment.duration(date.diff(now));
console.log(duration.days())
console.log(duration.hours())
```
Please refactor.

Reply:
Refactored